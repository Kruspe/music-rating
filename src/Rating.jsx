import React, { useState } from 'react';
import { Rating as RatingMaterialUI } from '@material-ui/lab';

const Rating = () => {
  const [rating, setRating] = useState(0);
  return (
    <RatingMaterialUI value={rating} onChange={(event, value) => setRating(value)} />
  );
};

export default Rating;
