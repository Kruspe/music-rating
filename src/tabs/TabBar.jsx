import React, { useState } from 'react';
import { Tabs, Tab } from '@material-ui/core';
import Overview from './overview';
import EstimateWacken from './wacken';

const TabBar = () => {
  const [tab, setTab] = useState(0);
  return (
    <>
      <Tabs
        value={tab}
        indicatorColor="primary"
        variant="fullWidth"
        onChange={(event, value) => setTab(value)}
      >
        <Tab label="Overview" />
        <Tab label="Estimate Wacken" />
      </Tabs>
      {tab === 0 && <Overview />}
      {tab === 1 && <EstimateWacken />}
    </>
  );
};

export default TabBar;
