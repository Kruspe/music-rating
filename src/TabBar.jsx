import React, { useState } from 'react';
import { Tabs, Tab } from '@material-ui/core';
import Overview from './overview';
import Rating from './rating';
import EstimateWacken from './estimation/wacken';

const TabBar = () => {
  const [tab, setTab] = useState(0);
  return (
    <>
      <Tabs
        value={tab}
        indicatorColor="primary"
        variant="fullWidth"
        onChange={(event, value) => setTab(value)}
      >
        <Tab label="Overview" />
        <Tab label="Rate" />
        <Tab label="Estimate Wacken" />
      </Tabs>
      {tab === 0 && <Overview />}
      {tab === 1 && <Rating />}
      {tab === 2 && <EstimateWacken />}
    </>
  );
};

export default TabBar;
