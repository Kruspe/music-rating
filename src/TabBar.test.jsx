import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Tabs } from '@material-ui/core';
import TabBar from './TabBar';
import Overview from './overview';
import Rating from './rating';

describe('TabBar', () => {
  it('should render correctly', () => {
    expect(toJSON(shallow(<TabBar />))).toMatchSnapshot();
  });

  it('should show overview as initial tab', () => {
    const wrapper = shallow(<TabBar />);
    expect(wrapper.find(Overview)).toHaveLength(1);
    expect(wrapper.find(Rating)).toHaveLength(0);
  });

  describe('Tabs', () => {
    it('should change tab to Overview when index is 0', () => {
      const wrapper = shallow(<TabBar />);
      wrapper.find(Tabs).prop('onChange')({}, 1);
      expect(wrapper.find(Rating)).toHaveLength(1);
      wrapper.find(Tabs).prop('onChange')({}, 0);
      expect(wrapper.find(Rating)).toHaveLength(0);
      expect(wrapper.find(Overview)).toHaveLength(1);
    });
    it('should change tab to Rating when index is 1', () => {
      const wrapper = shallow(<TabBar />);
      expect(wrapper.find(Overview)).toHaveLength(1);
      wrapper.find(Tabs).prop('onChange')({}, 1);
      expect(wrapper.find(Rating)).toHaveLength(1);
      expect(wrapper.find(Overview)).toHaveLength(0);
    });
  });
});
