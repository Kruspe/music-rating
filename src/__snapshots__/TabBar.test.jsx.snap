// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBar should render correctly 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Tabs))
    indicatorColor="primary"
    onChange={[Function]}
    value={0}
    variant="fullWidth"
  >
    <WithStyles(ForwardRef(Tab))
      label="Overview"
    />
    <WithStyles(ForwardRef(Tab))
      label="Rating"
    />
  </WithStyles(ForwardRef(Tabs))>
  <Overview />
</Fragment>
`;
